packages/components/src/components/modals/Modal/Modal.tsx

```diff
@@ -97,7 +99,7 @@ const Heading = styled(H3)<HeadingProps>`
                 margin-right: ${spacingsPx.xs};
             }
         `}
-    ${({ $headingSize }) => HEADING_SIZES[$headingSize ?? 'default'].css};
+    ${({ $headingSize = 'default' }) => HEADING_SIZES[$headingSize].css};
```

Comment:
Nit but you can remove the default value here if you set `$headingSize` a required.